### ISSUE TITLE
SwiftMailer v6 problems

### ISSUE THREAD

**Issue by Mozan** [contao/core-bundle#843]
Hello,

While upgrading one of my sites from v4.2.5 to the current (standard-edition - v4.3.10) I found out that pretty much everything related to emails via the "**mail**" transport do not work.

PHP - v7.0.x 
Below you'll find my actual composer.json `require` part:
```
    "require": {
        "php": ">=5.5.0",
        "symfony/symfony": "^2.8|^3.0",
        "symfony/monolog-bundle": "^2.8",
        "symfony/swiftmailer-bundle": "^2.3",
        "sensio/distribution-bundle": "^5.0",
        "sensio/framework-extra-bundle": "^3.0.2",
        "doctrine/doctrine-bundle": "^1.6",
        "lexik/maintenance-bundle": "^2.0",
        "contao/core-bundle": "^4.3.1",
        "contao/installation-bundle": "^4.3.5",
        "incenteev/composer-parameter-handler": "^2.0",
        "oneup/uploader-bundle": "~1.4",
        "terminal42/notification_center": "^1.3",
        "codefog/contao-widget_tree_picker": "^2.3",
        "phpoffice/phpexcel": "^1.8",
        "nelmio/api-doc-bundle": "^2.13",
        "lexik/jwt-authentication-bundle": "^2.0",
        "gfreeau/get-jwt-bundle": "2.0.x-dev",
        "terminal42/contao-fineuploader": "^2.0",
        "nelmio/cors-bundle": "^1.4"
    }
```

and, my `app/config/parameters.yml` related to mail
```
    mailer_transport: mail
    mailer_host: 127.0.0.1
    mailer_user: null
    mailer_password: null
    mailer_port: 25
    mailer_encryption: null
```

**So - where's the problem?**
**First one:**
in `core-bundle/src/Resources/contao/library/Contao/Email.php`
around line 120 we have such code:
```
public function __construct()
{
    $this->strCharset = \Config::get('characterSet');

    // Instantiate Swift_Message
    $this->objMessage = \Swift_Message::newInstance();
    $this->objMessage->getHeaders()->addTextHeader('X-Mailer', 'Contao Open Source CMS');
}
```
There is no more `\Swift_Message::newInstance()` method in v6.

**Second:** 
as per http://stackoverflow.com/questions/41426486/the-swift-transport-mailtransport-class-is-deprecated-since-version-5-4-5-and-wi
It's true. v6 version removed the mail transport (`Swift_Transport_MailTransport`).
In the `composer update` process Contao suggest to use the **mail** transport which is from now - simply wrong.

**Suggested fixes - as always - no time to create a PR - would love to, maybe next time:**
**Ad 1:**
`core-bundle/src/Resources/contao/library/Contao/Email.php`
```
public function __construct()
{
    $this->strCharset = \Config::get('characterSet');

    // Instantiate Swift_Message
    //$this->objMessage = \Swift_Message::newInstance();
    $this->objMessage = new \Swift_Message();
    $this->objMessage->getHeaders()->addTextHeader('X-Mailer', 'Contao Open Source CMS');
}
```

**Ad 2:**
Fix is simple:
`app/config/parameters.yml`
  ```mailer_transport: sendmail```

However I have no idea how to fix it in the composer process. I think it's an easy one. Just suggest to use `sendmail` instead of `mail`?

**Comment by leofeyer** [comment]
How did you manage to install `swiftmailer/swiftmailer` in version 6? We are using version 5 and Composer should not update it to a new major version without explicit configuration changes.

The standard edition currently uses `symfony/swiftmailer-bundle v2.5.4`, which requires `swiftmailer/swiftmailer (>=4.2.0,~5.0)`, and according to packagist.org the requirement only changes in version `v3.0.0` of the bundle:

https://packagist.org/packages/symfony/swiftmailer-bundle#v2.6.2

**Comment by fritzmg** [comment]
@leofeyer 
```
- symfony/swiftmailer-bundle v2.6.1 requires swiftmailer/swiftmailer >=4.2.0|~5.0 -> satisfiable by swiftmailer/swiftmailer[v6.0.1].
```
Das requirement `>=4.2.0|~5.0` bedeutet ja im Endeffekt _alles ab 4.2.0_. Die Angabe `|~5.0` ist eigentlich überflüssig. Bei einem composer update in einer Contao 4 Installation wird auf jeden Fall Swiftmailer 6 installiert, vorausgesetzt das PHP CLI oder die PHP platform requirement ist Version 7 oder höher.

**Comment by leofeyer** [comment]
> The requirement `>=4.2.0|~5.0` means that _everything from 4.2.0 on_ can be installed. The usage of `|~5.0` is actually superfluous.

I don't think so.

> When you execute a composer update in a Contao 4 installation, Swiftmailer 6 will definitely be installed, provided that your PHP CLI or your PHP platform requirement is version 7 or higher.

Nope, not the case here.

@contao/developers /cc

**Comment by fritzmg** [comment]
> Nope, not the case here.

The aforementioned message
```
- symfony/swiftmailer-bundle v2.6.1 requires swiftmailer/swiftmailer >=4.2.0|~5.0 -> satisfiable by swiftmailer/swiftmailer[v6.0.1].
```
was taken from the composer output of an unsuccessful `composer update` with PHP 5.6.

**Comment by fritzmg** [comment]
> I don't think so.

It is a bug that was introduced in `symfony/swiftmailer-bundle` `2.6.0` and still present in `2.6.1` and it was fixed in `2.6.2`. See

https://packagist.org/packages/symfony/swiftmailer-bundle#v2.5.5 vs.
https://packagist.org/packages/symfony/swiftmailer-bundle#v2.6.0 vs.
https://packagist.org/packages/symfony/swiftmailer-bundle#v2.6.1 vs.
https://packagist.org/packages/symfony/swiftmailer-bundle#v2.6.2

**Comment by aschempp** [comment]
so we should update our dependency to swiftmailer-bundle 2.6.2 ?

**Comment by fritzmg** [comment]
> so we should update our dependency to swiftmailer-bundle 2.6.2 ?

Or add `2.6.0` and `2.6.1` as a conflict.

**Comment by ausi** [comment]
This is the related bug regarding the buggy version constraint: symfony/swiftmailer-bundle#174

I agree with @fritzmg, we should keep the dependency as it is and add a conflict with `2.6.0` and `2.6.1`.

**Comment by leofeyer** [comment]
Fixed in a4999a370895f64432d0a684b7babb22f2aba014.
